# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextGLX.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/gl/GLContextGLX.h
--- a/gfx/gl/GLContextGLX.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/gl/GLContextGLX.h	Wed Jun 10 22:21:02 2020 +0000
@@ -20,9 +20,8 @@
  public:
   MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(GLContextGLX, override)
   static already_AddRefed<GLContextGLX> CreateGLContext(
-      CreateContextFlags flags, const SurfaceCaps& caps, bool isOffscreen,
-      Display* display, GLXDrawable drawable, GLXFBConfig cfg,
-      bool deleteDrawable, gfxXlibSurface* pixmap);
+      const GLContextDesc&, Display* display, GLXDrawable drawable,
+      GLXFBConfig cfg, bool deleteDrawable, gfxXlibSurface* pixmap);
 
   static bool FindVisual(Display* display, int screen, bool useWebRender,
                          bool useAlpha, int* const out_visualId);
@@ -66,8 +65,7 @@
  private:
   friend class GLContextProviderGLX;
 
-  GLContextGLX(CreateContextFlags flags, const SurfaceCaps& caps,
-               bool isOffscreen, Display* aDisplay, GLXDrawable aDrawable,
+  GLContextGLX(const GLContextDesc&, Display* aDisplay, GLXDrawable aDrawable,
                GLXContext aContext, bool aDeleteDrawable, bool aDoubleBuffered,
                gfxXlibSurface* aPixmap);
 